scripts/portfolio.js

```diff
@@ -1,30 +1,63 @@
-var portfolioProjects = [];
-
 function Project (opts) {
-  this.title = opts.title;
-  this.category = opts.category;
-  this.projectUrl = opts.projectUrl;
-  this.publishedOn = opts.publishedOn;
-  this.featureImage = opts.featureImage;
-  this.body = opts.body;
+  for (keys in opts) {
+    this[keys] = opts[keys];
+  }
 }
+//Attach to the constructor, not global
+Project.all = [];
 
-Project.prototype.toHtml = function() {
-  var $source = $('#project-render').html();
-  var compiledProject = Handlebars.compile($source);
+Project.prototype.toHtml = function(templateId) {
+  var template = Handlebars.compile((templateId).html());
   this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
   this.publishStatus = this.publishedOn ? + this.daysAgo + ' days ago' : '(draft)';
-  return compiledProject(this);
+  return template(this);
 };
 
 // Sort our data by date published, descending order
-portfolioProjects.sort(function(a,b) {
-  return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-});
+Project.loadAll = function(passedData) {
+  passedData.sort(function(a,b) {
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  });
+  passedData.forEach(function(ele) {
+    Project.all.push(new Project(ele));
+  });
+};
+
 
-myProjects.forEach(function(ele) {
-  portfolioProjects.push(new Project(ele));
-});
-portfolioProjects.forEach(function(a){
-  $('#projects').append(a.toHtml());
-});
+//reusable function to render from JSON and set localStorage for next loadAll
+var renderFromJSON = function() {
+  $.getJSON('data/portfolioData.json', function(data) {
+    Project.loadAll(data);
+    localStorage.portfolioData = JSON.stringify(data);
+    portfolioView.initIndexPage();
+    $.ajax({
+      type: 'GET',
```

Comment:
You don't need to grab all the data twice to get the eTag. You can either just get the headers, or you can pass in data, msg, xhr to the $.getJSON callback function.